### ISSUE TITLE
Add internationalization support

### ISSUE THREAD

**Issue by unixjazz** [antievictionmappingproject/covid-19-map#16]
I would like help w/ translations! I was wondering what library / framework you would like to use to facilitate the internationalization of the project?

Some of the translation work started here on various languages (for the form + interface):
[Wikimedia Etherpad: Emergency Tenant Protection Intl](https://etherpad.wikimedia.org/p/emergency-tenant-protection-intl)

(using a pad, however, is not ideal...)

**Comment by clhenrick** [comment]
Hi @unixjazz! Thanks for the offer and for starting this work. 

One piece that makes this somewhat complicated is that we'll need to handle translation of crowd sourced submissions (the info that appears after clicking on a polygon or circle in the map). This could either be done manually in the Google Sheet that contains the form responses, or on the perhaps on the fly using something like Google Translate.

We're open to suggestions for a library or framework as well for the UI.

**Comment by erinmcel** [comment]
Amazing! We also have another form we need translated for rent strikes if you're game :)

**Comment by erinmcel** [comment]
Chinese version of the form here: https://pad.riseup.net/p/_5O-cw9DTBD7gsl-w3GT-keep

**Comment by unixjazz** [comment]
Hi @clhenrick, @erinmcel: I hear you, internationalization is far from trivial. I really wish I could help more, but I am not a web developer myself. [Here](https://github.com/unixjazz/covid-19-map/commit/ea08ba974b498f3ad2f93f23de7a3bb77b3e3c06) I tried to take the easy (and quick) route by redirecting the user to the translated page + form based on the [browser locale](https://unixjazz.github.io/covid-19-map/pt-br/index.html). But... this approach is far from consistent.

For larger projects, people use gettext + [poeditor](https://poeditor.com/) (for translation of .po files). I do not think it would be the case here, since just a few strings need to be translated. Maybe something like '[localeplanet](https://www.localeplanet.com/support/howto-localize-js.html)' would do?

p.s. I would be glad to continue to help with the translation!

**Comment by erinmcel** [comment]
Here is Portuguese and German for the rent strike form: https://pad.riseup.net/p/_5O-cw9DTBD7gsl-w3GT-keep

**Comment by clhenrick** [comment]
Thanks @unixjazz for the example and suggestions. I think we can (and should) avoid duplicating the HTML structure by using a JS library like [i18next](https://www.i18next.com). Even though our site isn't too complex now, I think it's important to choose a solution that helps it scale as it grows in complexity.

**Comment by unixjazz** [comment]
@clhenrick: I totally agree! You need something that can scale for the global map. I think there is a similar problem with respect to Google forms: you would have to create a different form for each translation (otherwise you will end up overloading the UI).

I have been thinking about this problem. I will report here as soon as I have something to share.

**Comment by erinmcel** [comment]
Paul is now going to create individual google forms for each language. So this can be a temporary fix.

**Comment by clhenrick** [comment]
For the "v1" of adding i18n using @unixjazz's translation JSON, we may want to use the [Handlebars.JS](https://handlebarsjs.com) library as it supports [pre-compilation of templates](https://handlebarsjs.com/api-reference/compilation.html#handlebars-compile-template-options) which will make the language toggling a bit more performant.

Currently we're using MustacheJS for the popup and infowindow HTML templates, which has an almost identical syntax to Handlebars but doesn't have as many features.

**Comment by jimboweb** [comment]
Another i18n issue that's not language-related is right now the map always loads with every layer but nations showing. It might be better to see where people are coming from. If they're outside it might be better to show different layers, for example show nations, cities and rent strikes and hide counties and maybe states Ideally we'd only show states if they're in a country where we have state data.

**Comment by clhenrick** [comment]
@jimboweb that's a good point, though for setting the language of the page AFAIK it's a better practice to use the browser's language setting rather than the user's location. This is because someone may be located in a country but does not speak that country's native language or does not prefer that language when they browse the web.

**Comment by unixjazz** [comment]
@clhenrick: I am glad you mentioned the template system! I did not know it was on the table to substitute it. If we can use Handlebars, we can solve one of the problems I encountered when trying to dynamically translate text in Leaflet popups. @river-honer and I will work on this, document our suggestion, and report back.

@jimboweb: I see your point. I am still not entirely sure if we can have one deployment for all locations / countries...  one suggestion I made was to have different deployments per region, modifying the queries to load only the data that is relevant for a region / country. I am anticipating that having global data points (on various layers) will slow down the map for everyone. This is up for future discussion.
